account_invoice_triple_discount/models/account_invoice.py

```diff
@@ -60,7 +67,7 @@ def triple_discount_preprocess(self):
                 discount2=line.discount2,
                 discount3=line.discount3,
             )
-            line.update({
+            line._cache.update({
```

Comment:
Why do you deal with the cache? Is it required. Behind the scene it's what the `update`method do

Reply:
@lmignon I have to deal with the cache because (as the module is currently designed), every discount fields are computed into the standard discount field (into cache). Then, after calling the standard behaviour (`super`) the custom `discount` fields are re-updated with filled value.
But as there is `api.multi` and `api.one`, Odoo change the cache in some case.
I agree with your previous comment: maybe it's a design problem or an issue into Odoo core (so in this case, my fix is only a workaround).